go/bindings/metrics.go

```diff
@@ -0,0 +1,84 @@
+package bindings
+
+/*
+#include "../../crates/bindings/flow_bindings.h"
+*/
+import "C"
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// A prometheus.Collector that exposes general rust metrics that are exposed by libbindings as
+// prometheus metrics.
+type promCollector struct{}
+
+// Describe implements prometheus.Collector for promCollector
+func (c *promCollector) Describe(ch chan<- *prometheus.Desc) {
+	prometheus.DescribeByCollect(c, ch)
+}
+
+// Collect implements prometheus.Collector for promCollector
+func (c *promCollector) Collect(ch chan<- prometheus.Metric) {
+	var stats = C.get_memory_stats()
+	var gauge = func(desc *prometheus.Desc, value C.uint64_t) {
+		ch <- prometheus.MustNewConstMetric(
+			desc,
+			prometheus.GaugeValue,
+			float64(value),
+		)
+	}
+	gauge(activeBytesDesc, stats.active)
+	gauge(allocBytesDesc, stats.allocated)
+
+	gauge(mappedBytesDesc, stats.mapped)
+	gauge(metadataBytesDesc, stats.metadata)
+
+	gauge(residentBytesDesc, stats.resident)
+	gauge(retainedBytesDesc, stats.retained)
+}
+
+var promCollectorInstance = &promCollector{}
+
+var registration = sync.Once{}
+
+// RegisterPrometheusCollector initializes prometheus metric collection that's pulled from the rust
+// libbindings. Subsequent calls to this function will panic.
+func RegisterPrometheusCollector() {
+	prometheus.MustRegister(promCollectorInstance)
+}
+
+var (
+	activeBytesDesc = prometheus.NewDesc(
+		"flow_rust_mem_active_bytes",
```

Comment:
Just confirming, do these names align with the Go memory stats ?

Reply:
No. Jemalloc exposes stats that are kind of specific to jemalloc, so I tried to keep naming of these metrics fairly consistent with the names used by the `stats._` properties from the [jemalloc man page](http://jemalloc.net/jemalloc.3.html). These are the same names used in the `GlobalMemoryStats` struct. I don't honestly see a lot of benefit to having these match the go metric names, since they're mostly tracking different things.